### ISSUE TITLE
Non-modal dialog does not appear as child of parent document

### ISSUE THREAD

**Issue by PICARDBRB** [nvaccess/nvda#13733]
Maybe related to https://github.com/nvaccess/nvda/pull/8194.  Modeless and Modal dialogs behave correctly with browse mode navigation, however, initiating the parent document tree with NVDA key plus F7 key for modeless aria-modal="false", the document tree is missing the dialog elements alongside the parent.

Steps to reproduce:
Using the html test case: https://github.com/nvaccess/nvda/files/1930487/dialog-modal-modeless.html.txt
Tested modal and non-modal dialogs in Chrome, Edge.
Open the testcase in the browser
Actual behavior: For Non Modal:
1. in BrowseMode with Modeless button activated
2. NVDA key plus F7
3. Review buttons tab
4. Ok button is listed

Expected behavior: for Non Modal
1. in BrowseMode with Non Modal button activated
2. NVDA key plus F7
3. Review buttons tab
4. Ok, Modal, and Modeless buttons are listed

Actual and Expected behavior: For Modal:
1. in BrowseMode with Modal button activated
2. NVDA key plus F7
3. Review buttons tab
4. Ok button is listed

System configuration
NVDA version: 2021.3.3
Windows version: 10.01.19042
Name and version of other software in use when reproducing the issue: 
- Chrome Version 101.0.4951.67 (Official Build) (64-bit)
- MS Edge Version 101.0.1210.47 (Official build) (64-bit)

Does the issue still occur after restarting your computer? Yes

thanks in advance.

**Comment by michaelDCurran** [comment]
At the moment this behavior is deliberate. When opening either dialog, NVDA's browse mode is trapped within the dialog, and therefore arrowing up and down only allows accessing the content of the dialog, similarly, NVDA+f7 only lists elements within the dialog.
Perhaps there is an argument that for modless dialogs, NvDA should not place the content in its own browse mode document limited to the dialog, but instead, render the modeless dialog content within the parent browse mode document. This would then mean that NVDA+f7  would list all the buttons as you expect.
I think this requires further understanding of the ARIA spec and debate about exactly what NVDA users want/expect here.

**Comment by PICARDBRB** [comment]
Hi, appears NVDA is not trapped within the Modeless dialog for browse mode.  User can navigate with up and down arrow keys outside of the modeless dialog and into the parent document.  NVDA should be consistent in behaviour, where author allows navigation outside of the dialog using appropriate aria-modal value, the screen reader should be consistent and list as a component of the parent document. 

Note additional JAWS test case:
Actual and Expected behavior: For Non Modal:
in BrowseMode with NonModal button activated
JAWS key plus F5 (form elements list)
Review buttons
Ok, Modal, and Modeless buttons are listed

System:
Windows version: 10.01.19042
MS Edge Version 101.0.1210.47 (Official build) (64-bit)
JAWS 2020.

It appears you are seeing a different behaviour for modeless dialog in browse mode and arrow keys?

**Comment by myasonik** [comment]
With the HTML dialog element gaining broader adoption now and with it having separate APIs to show modal vs non-modal dialogs, better support for non-modal dialogs seems more important now. Like @PICARDBRB says, at least consistency between how browse mode and how the element list treats them seems important. Potentially announcing them differently makes sense as well? Not sure.

Here's a [simple test case in Codepen](https://codepen.io/myasonik/pen/GReoGYw) where two different buttons (using two different APIs) open the same dialog. The first usage should be modal, blocking access to the original page. The second usage should be non-modal, maintaining access to the original page. 

Talkback+Chrome and VoiceOver+Safari work as expected.

<hr>

As a note, I'll mention I don't use NVDA or any screen reader day-to-day and only bring this up as a web dev.
